packages/@aws-cdk/custom-resources/lib/provider-framework/provider.ts

```diff
@@ -0,0 +1,198 @@
+import iam = require('@aws-cdk/aws-iam');
+import lambda = require('@aws-cdk/aws-lambda');
+import sfn = require('@aws-cdk/aws-stepfunctions');
+import tasks = require('@aws-cdk/aws-stepfunctions-tasks');
+import { Construct, Duration } from '@aws-cdk/core';
+import path = require('path');
+import consts = require('./runtime/consts');
+import { calculateRetryPolicy } from './util';
+
+const RUNTIME_HANDLER_PATH = path.join(__dirname, 'runtime');
+
+/**
+ * Initialization properties for the `Provider` construct.
+ */
+export interface ProviderProps {
+  /**
+   * The handler JavaScript code for the custom resource (Node.js 10.x).
+   *
+   * Must contain the handler files with the approriate exports as defined by `onEventHandler`
+   * and `isCompleteHandler`. The defaults are `index.onEvent` and `index.isComplete`.
+   */
+  readonly code: lambda.Code;
+
+  /**
+   * The AWS Lambda runtime to use. Currently, only NodeJS runtimes are supported.
+   */
+  readonly runtime: lambda.Runtime;
+
+  /**
+   * The function to invoke for all resource lifecycle operations
+   * (CREATE/UPDATE/DELETE).
+   *
+   * This function is responsible to begin the requested resource operation
+   * (CREATE/UPDATE/DELETE) and return any additional properties to add to the
+   * event, which will later be passed to `isComplete`. The `PhysicalResourceId`
+   * property must be included in the response.
+   */
+  readonly onEventHandler: string;
+
+  /**
+   * The function to invoke in order to determine if the operation is
+   * complete.
+   *
+   * This function will be called immediately after `onEvent` and then
+   * periodically based on the configured query interval as long as it returns
+   * `false`. If the function still returns `false` and the alloted timeout has
+   * passed, the operation will fail.
+   */
+  readonly isCompleteHandler: string;
+
+  /**
+   * Policy statements to add to the IAM role that executes the user handlers.
+   *
+   * The `Provider` class also implements `iam.IGrantable`, which means you can
+   * use it as a target of `grantFoo` methods (e.g.
+   * `bucket.grantRead(provider)`).
+   *
+   * Bear in mind that usually there is a single provider which handles
+   * provisioning for multiple resources, so it's execution role should allow it
+   * to manage multiple resources of the same type.
+   *
+   * @default - empty policy
+   */
+  readonly policy?: iam.PolicyStatement[];
```

Comment:
This prop is not used to make `addToPolicy()` calls

 (the integ test uses a `grantXxx()` method)

Reply:
Nice catch, thanks